### PR TITLE
MAGN-6145 Create a conversion node based on the existing prototype.

### PR DESCRIPTION
**Pull request by ramramps**
The task is to create a new Conversion node (based on an existing prototype).

The new conversion node will convert between basic unit types that can multiple units can be worked on the same graph. 

To execute this branch, please import the library  - DynamoConversions.dll (in the debug folder). 

Note: The equal (=) button on the node will toggle the unit types. 

- [ ] @ikeough

### CODE REVIEW

**Comment by ikeough** [comment]
@ramramps Can you set the message of your commits according to what they do? All of your commits have the message "Changes for dynamo conversion." It's just a little hard to navigate between them, and understand their intent (even though you've put that in the extended commit message).

**Comment by ikeough** [comment]
Please move the DynamoConversions project into the Libraries folder.

**Comment by ikeough** [comment]
We have to figure out how to kill this warning. First, we shouldn't try to cast a null to a double. I'm not sure how we prevent this in the AST, but we should try. And I'd like to know why these are piling up too. There seems to be one happening for every execution even with an input attached.

image

**Comment by ikeough** [comment]
I think we should get rid of all the margins around the dropdowns and the toggle button. The space around these elements and their offset from the position of the ports is distracting.

image

**Comment by ikeough** [comment]
Update the comments on the "Convert" node, so we don't reference SI units any more.

image

**Comment by ikeough** [comment]
@ramramps Lots to do here. Let me know as soon as you've made the updates and I'll take another look. Thanks!

File: src/DynamoConversions/Conversions.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            {ConversionUnit.Millimeters, .001},
+            {ConversionUnit.Centimeters, .01},
+            {ConversionUnit.Meters, 1.0},
+            {ConversionUnit.Degrees, 0.0174532925},
+            {ConversionUnit.Radians, 1},
+            {ConversionUnit.Pounds, 0.453592},
+            {ConversionUnit.Kilograms, 1}
+        };
+
+        public static double ConvertToSI(double value, double conversion, double conversionto)
```
Comment (ikeough) [inline src/DynamoConversions/Conversions.cs]: Change the name of this method. It's no longer converting anything to/from SI units.

File: src/Libraries/CoreNodeModels/DynamoConvert.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+        #region Serialization/Deserialization Methods
+
+        protected override void SerializeCore(XmlElement element, SaveContext context)
+        {
+            base.SerializeCore(element, context); // Base implementation must be called.
+
+            var xmlDocument = element.OwnerDocument;
+            var subNode = xmlDocument.CreateElement("Range");
+            subNode.SetAttribute("conversionFrom", SelectedFromConversion.ToString(CultureInfo.InvariantCulture));
```
Comment (ikeough) [inline src/Libraries/CoreNodeModels/DynamoConvert.cs]: What happens if SelectedFromConversion or SelectedToConversion is null?

**Comment by ikeough** [comment]
@ramramps Lots to do here. Let me know as soon as you've got the changes in and I'll take another look. It's not as bad as the number of comments make it look :)

**Comment by ramramps** [comment]
@ikeough  - I think I have addressed the changes as you mentioned, except for those warning messages. There has to be some check on the AstFactory when building two double nodes, I have to check with Jun for that.

**Comment by ikeough** [comment]
@ramramps a few more questions.

**Comment by ramramps** [comment]
@ikeough  - fixed the issues with localization. 
Margin of 1 is used to space the combo box and button. 
Removed the IsExecuted parameter. It is not needed anymore

**Comment by ikeough** [comment]
@ramramps LGTM. We'll probably need to fiddle with the UI a bit once it's in, but I don't want that to hold it.

**Pull request merged**